### PR TITLE
Bug 1195428 - [Music][NGA] Wire up front-end to playlist queues

### PR DESCRIPTION
**Pull request by justindarc**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9488854c65622d47bca8eb68aaf1ae0dc2b7339

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dfd88cbe85f05e8e9180f18d083d3802b675f1c5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d03a74703621a8174644c6662477b762a684193f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=322e0c27a7d64ba7f7dd874155feacb046b0552e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a601d1c4fa68c65adb16517a3827374d7e4a29b4

**Comment by hfiguiere** [comment]
I don't see where the playlist are defined.... They don't appear in the playlist tab either.

**Comment by justindarc** [comment]
@hfiguiere sorry I may have not been clear, this patch just hooks up the queues (albums, artists, all songs) as well as shuffle/repeat for the queues. This doesn't populate the "Playlists" tab. I'll land that separately.

File: dev_apps/music-nga/elements/music-artwork.js [hunk 1]
```diff
@@ -178,6 +213,14 @@ proto.attributeChangedCallback = function(attr, oldVal, newVal) {
   }
 };
 
+proto.nextRepeat = function() {
```
Comment (jimporter) [inline dev_apps/music-nga/elements/music-artwork.js]: `queue.js` has its own `Repeat.next()` function; maybe use that here?

File: dev_apps/music-nga/elements/music-rating.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  #container[data-value="2"] > button[data-value="1"],
+  #container[data-value="2"] > button[data-value="2"],
+  #container[data-value="3"] > button[data-value="1"],
+  #container[data-value="3"] > button[data-value="2"],
+  #container[data-value="3"] > button[data-value="3"],
+  #container[data-value="4"] > button[data-value="1"],
+  #container[data-value="4"] > button[data-value="2"],
+  #container[data-value="4"] > button[data-value="3"],
+  #container[data-value="4"] > button[data-value="4"],
+  #container[data-value="5"] > button {
```
Comment (jimporter) [inline dev_apps/music-nga/elements/music-rating.js]: You could probably use `-moz-any` here, if you wanted.

File: dev_apps/music-nga/elements/music-rating.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+`<style scoped>
+  #container {
+    display: flex;
+    flex-flow: row nowrap;
+    width: 100%;
+    height: 100%;
+  }
+  #container > button {
+    background-color: transparent;
+    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAQAAAC0NkA6AAAB2UlEQVR4Ad3XT0vjWhjA4Sfpv9gLQqlahTv7+f6fRcSO2lFHdGFxOtJ6m7Y2d5NFKBpyarKZ50A3hfx4zxso9deIBMk6vuGXtQDtTJB/DZD5KUAsRM8ADPSai5yKQOS0qUjHUM5Qp5nISCQnMmoi0nKs6Fir/siJWFHspO5I7KQ8W0fkSNuutqPaIiVrHonqiwx1faRrWF9kFP5NaGQg8ZnE4KuRSOLQmTJnDiUiJaKMorZefrp6uqpbSaX5Z2qzG2kbyB8tVo+tND+/baKM7/7RnIXLGIkmJcR40KQHYkzdyTQhc2dKDKZuG8hkbk3JI3gxqTmTmXihGGHm2lZdtq7N2I3w6sq7Ory78spHEeZhmZLEnM8iLIxtfMXGDwvKIvxnbG1fa2NvlEdgaWxlHytjS6pESPNpwqdIqRphZStUZkVIpKUnVFcrLHJgH/2wSN8+DhqfJDzSb/66Iol9JKLqkZ7YPmK96pE+1HdhceDal27cWIatPmSStXsXZmYu3FvXP8nWk3PPMpB5du7Jdv9J2joK8kc+2u5kH/NsUUe7WqSvKL+c0gss6leLJMXfydI1y1+FsQW5pFrkTYbUxKW5KuYuTaTIvJX9dSjq6/ojEyZyaF0Sadb/h/OY56tQNmIAAAAASUVORK5CYII=);
```
Comment (jimporter) [inline dev_apps/music-nga/elements/music-rating.js]: Is there a reason we're embedding the background-image like this? I find it harder to debug when I can't look at the CSS and know what the image should be (as I can if it's just a URL to a file on the disk somewhere).

File: dev_apps/music-nga/elements/music-artwork.js (hunk 1 above)
Comment (justindarc) [inline dev_apps/music-nga/elements/music-artwork.js]: Yeah, I saw that. I didn't want to cross-pollenate between a web component and the backend though.

File: dev_apps/music-nga/elements/music-rating.js (hunk 2 above)
Comment (justindarc) [inline dev_apps/music-nga/elements/music-rating.js]: Oh? I didn't know about that. I'll take a look, thanks!

File: dev_apps/music-nga/elements/music-rating.js (hunk 3 above)
Comment (justindarc) [inline dev_apps/music-nga/elements/music-rating.js]: I don't think so actually. Not sure what I was thinking when I did this. I'll switch this to use a real URL.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76f625740d2d604268a35a61f379d1ce9b0ea8ae

**Pull request merged**